src/SFA.DAS.Employer.PR.Web/Authentication/EmployerOwnerAuthorizationHandler.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using SFA.DAS.Employer.PR.Domain.OuterApi.Responses;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using System.Text.Json;
+
+namespace SFA.DAS.Employer.PR.Web.Authentication;
+
+public class EmployerOwnerAuthorizationHandler : AuthorizationHandler<EmployerOwnerRoleRequirement>
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public EmployerOwnerAuthorizationHandler(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, EmployerOwnerRoleRequirement requirement)
+    {
+        if (!IsEmployerAuthorised(context))
+        {
+            return Task.CompletedTask;
+        }
+
+        context.Succeed(requirement);
+
+        return Task.CompletedTask;
+    }
+
+    private bool IsEmployerAuthorised(AuthorizationHandlerContext context)
+    {
+        if (!_httpContextAccessor.HttpContext!.Request.RouteValues.ContainsKey(RouteValueKeys.EncodedAccountId))
+        {
+            return false;
+        }
+
+        var accountIdFromUrl = _httpContextAccessor.HttpContext.Request.RouteValues[RouteValueKeys.EncodedAccountId]!.ToString()!;
+        var employerAccountClaim = context.User.FindFirst(c => c.Type.Equals(EmployerClaims.AccountsClaimsTypeIdentifier));
+
+        if (employerAccountClaim?.Value == null)
+        {
+            return false;
+        }
+
+        Dictionary<string, EmployerUserAccountItem> employerAccounts;
+        EmployerUserAccountItem? employerAccount = null;
+
+        employerAccounts = JsonSerializer.Deserialize<Dictionary<string, EmployerUserAccountItem>>(employerAccountClaim.Value)!;
+
+        if (employerAccounts != null)
```

Comment:
You have null negated on line 47 and then checking for not null on line 49, it is always going to be not null else it would throw exception on line 47 itself, you can try this out.

Reply:
yes that's a mistake, fixed